### DISCUSSION TITLE
Best setup for raspberry pi, with docker, selenium and python

### DISCUSSION THREAD

**Discussion by syedammarali** [jdholtz/auto-southwest-check-in#136] [Q&A] [answered]
Hello everyone!

I need some help with a setup I am trying to put in place. I am very new to this so might ask very stupid 101 questions so please bear with me:

Background:

I had a Raspberry Pi which I primarily used for HomeBridge, with it directly installed on my Pi. I recently wrote a Python script using Selenium on Chrome to automate making reservations at my club. The script needs to be run on multiple schedules, so I thought I’d setup my Pi to automate/schedule the execution. I decided to use docker, as it would allow me to continue using the Pi for HomeBridge as well. 

I have successfully installed Docker and Portainer on DietPi, and have HomeBridge up and running. 

Objective
Now for the other use case. I want to get python and selenium up and running as well and this is where I am confused. I’ve installed the selenium multi-arch grid in a container, and now am struggle to understand how the remaining works. I’ve been also trying to use VSCode to remotely use my computer, but that might be step 2.

Question

1) I have pulled the python image on my Pi, and it shows on portioner as unused. What I am trying to first understand is how this works. I’ve seen multiple videos but it’s confusing as to whether I need to attach python to the selenium container, or they’re individual containers. Or I use Pi native python with selenium in a container

The confusion is that whatever instruction videos I find, assume docker is installed locally. So I am not sure what to do.

2) I understand that I can use VSCode to remotely connect to the docker. In this setup, does the python I use need to be the Pi native python or the container based python? Do I still need to install docker on the “dev” machine, for it to connect to the docker on my Pi?

**Comment by jdholtz** [comment, marked as answer]
Hi. This discussion board is meant for questions and conversations about Auto-Southwest Check-In, so you'll probably get much better responses if you ask on Reddit, StackOverflow, etc.

As for 1, if I understand it correctly, you can have a similar setup that this project has. Your container can be the Python base image, and you can install Selenium and other dependencies needed in the container. That would give you one container to have everything you need.

For 2, I don't have any experience with this, but you can probably setup your container to listen to a port on the Host (your Pi) and use any remote connection (SSH, RDP, etc.) to connect directly to your container. Once you are in the container, everything you need, such as Python, will already be in it so you should be good.
